pkg/controller/migassetcollection/validation.go

```diff
@@ -28,39 +28,48 @@ const (
 )
 
 // Validate the asset collection resource.
-// Returns error and the total error conditions set.
-func (r ReconcileMigAssetCollection) validate(assetCollection *migapi.MigAssetCollection) (int, error) {
-	totalSet := 0
+func (r ReconcileMigAssetCollection) validate(assetCollection *migapi.MigAssetCollection) error {
+	assetCollection.Status.BeginStagedConditions()
 
 	// Empty collection
-	nSet, err := r.validateEmpty(assetCollection)
+	err := r.validateEmpty(assetCollection)
 	if err != nil {
-		return 0, err
+		return err
+	}
+
+	// Apply changes
+	assetCollection.Status.EndStagedConditions()
```

Comment:
Maybe a better name would be `WriteStagedConditions()` or `ApplyStagedConditions()` or `CommitStagedConditions()`. `End` doesn't seem obvious to me.

Reply:
Using `End` because it _ends_ `Conditions` staging mode.

Perhaps instead of `Conditions.staging` _bool_, `Conditons.mode` _string_ (Normal|Staging) would be clearer?  This seems to be causing confusion.

Reply:
"Using End because it ends Conditions staging mode."
^ This makes sense to me. Would prefer we call the func `EndStaging` rather than `EndStaged`.

I like `Conditions.mode (normal|staging)` as you suggest!

Reply:
On 2nd thought, a binary mode may be overkill.  I like the _staging_ naming suggestion though.  How about we go with `Conditions.staging` and `BeginStagingConditions`, `EndStagingConditions`?

Reply:
@jortel sounds good